frontend/src/metabase/search/components/SearchResult/use-info-text.tsx

```diff
@@ -0,0 +1,155 @@
+import { useMemo } from "react";
+import { t } from "ttag";
+import type { WrappedResult } from "metabase/search/types";
+import type { Collection } from "metabase-types/api";
+import { useDatabaseQuery, useTableQuery } from "metabase/common/hooks";
+import {
+  browseDatabase,
+  browseSchema,
+  collection as collectionUrl,
+  tableRowsQuery,
+} from "metabase/lib/urls";
+import {
+  PLUGIN_COLLECTION_COMPONENTS,
+  PLUGIN_COLLECTIONS,
+} from "metabase/plugins";
+import type TableType from "metabase-lib/metadata/Table";
+
+export type InfoTextData = {
+  link?: string | null;
+  icon?: JSX.Element | null;
+  label: string | null;
+};
+
+const { CollectionAuthorityLevelIcon } = PLUGIN_COLLECTION_COMPONENTS;
+
+export const useInfoText = (result: WrappedResult): InfoTextData[] => {
```

Comment:
Instead of creating a bunch of different components with `InfoText.jsx`, I moved the logic into a `use-info-text` hook that we can use to get the link + label data. Then, we can inject that information into a new `InfoText.tsx` component that will display data with the same component. Essentially we're separating the data from the styling logic